src/convertToObject.js

```diff
@@ -6,7 +6,23 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const cleanedSourceString = sourceString.trim().replace(/;$/, '');
+
+  const result = {};
+
+  const rules = cleanedSourceString.split(';');
+
+  rules.forEach((rule) => {
+    if (rule.trim()) {
+      const [key, value] = rule.split(':').map((part) => part.trim());
+
+      if (key && value !== undefined) {
```

Comment:
The condition `value !== undefined` is technically correct, but it might be more explicit to check if `value` is not an empty string. This would prevent adding keys with empty values to the result object.